packages/nimble-components/src/table-column/base/models/column-validator.ts

```diff
@@ -0,0 +1,55 @@
+import type { ValidityObject } from '../../../table/types';
+import type { ColumnInternals } from './column-internals';
+
+type ObjectFromList<T extends readonly string[], V = string> = {
+    [K in T extends readonly (infer U)[] ? U : never]: V;
```

Comment:
Could we replace `U` with `string` since that is the type we're dealing with here? Is there a reason for it to be inferred?

Reply:
No, because we're actually passing an array of specific constants, so that when the type is inferred, it infers that specific union of constants. E.g. `ObjectFromList<['a', 'b', 'c'] as const>` => `U` is inferred as `'a' | 'b' | 'c'`.